examples/anthos-cicd-with-gitlab/docs/3-set-up-anthos-config-management.md

```diff
@@ -0,0 +1,323 @@
+cd prod
+cat > namespace.yaml << EOF
+apiVersion: v1
+kind: Namespace
+metadata:
+ name: prod
+ annotations:
+   configmanagement.gke.io/cluster-selector: prod-cluster-selector
+EOF
+```
+
+
+
+## Policy constraint
+
+Lastly, we’ll take advantage of the policy control feature of ACM and create a policy. [Policy controller](https://cloud.google.com/anthos-config-management/docs/concepts/policy-controller) is used to check, audit and enforce your clusters’ compliances with policies that may be related to security, regulations or arbitrary business rules. 
+
+In this tutorial we’ll create a no privileged container constraint and apply it to the prod namespace. Notice we set policycontroller to true in `config-management-prod.yaml `earlier. This will enable this policy to be enforced in our clusters.
+
+Create `constraint-restrict-privileged-container.yaml` in `cluster/` :
+
+
+```
+cd ~/$GROUP_NAME/acm/cluster
+
+cat > constraint-restrict-privileged-container.yaml << EOF
+apiVersion: templates.gatekeeper.sh/v1beta1
+kind: ConstraintTemplate
+metadata:
+ name: noprivilegedcontainer
+ annotations:
+    configmanagement.gke.io/cluster-selector: prod-cluster-selector
+spec:
+ crd:
+   spec:
+     names:
+       kind: NoPrivilegedContainer
+ targets:
+   - target: admission.k8s.gatekeeper.sh
+     rego: |
+       package noprivileged
+       violation[{"msg": msg, "details": {}}] {
+           c := input_containers[_]
+           c.securityContext.privileged
+           msg := sprintf("Privileged container is not allowed: %v, securityContext: %v", [c.name, c.securityContext])
+       }
+       input_containers[c] {
+           c := input.review.object.spec.containers[_]
+       }
+       input_containers[c] {
+           c := input.review.object.spec.initContainers[_]
+       }
+       input_containers[c] {
+           c := input.review.object.spec.template.spec.containers[_]
+       }
+       input_containers[c] {
+           c := input.review.object.spec.template.spec.initContainers[_]
+       }
+---
+apiVersion: constraints.gatekeeper.sh/v1beta1
+kind: NoPrivilegedContainer
+metadata:
+ name: no-privileged-container
+ annotations:
+    configmanagement.gke.io/cluster-selector: prod-cluster-selector
+spec:
+ enforcementAction: dryrun
+ match:
+   kinds:
+     - apiGroups: ["*"]
+       kinds: ["Deployment", "Pod"]
+EOF
+```
+
+
+
+## Push changes to remote
+
+Now that we have made these changes locally, let’s push these changes to the remote repo so the clusters can read from it and configure itself:
+
+Push changes to remote:
+
+
+```
+git add -A
+git commit -m "Set up ACM"
+git push
+```
+
+
+Since the dev cluster is configured to sync with the dev branch, create a new branch called `dev`  from the `main` branch by clicking the “+” sign in the repo page.
+
+ Apply the configuration:
+
+
+```
+for i in "dev" "prod"; do
+   gcloud container clusters get-credentials ${i} \
+       --zone=$ZONE
+   kubectl apply -f config-management-${i}.yaml
```

Comment:
I am getting this error here (shown only for one of the two setups):

```
error: error validating "config-management-dev.yaml": error validating data: [unknown object type "nil" in ConfigManagement.metadata, unknown object type "nil" in ConfigManagement.spec]; if you choose to ignore these errors, turn validation off with --validate=false
```

If I add `--validate=false` to `kubectl apply`, I obtain this error:

```
error: error when retrieving current configuration of:
Resource: "configmanagement.gke.io/v1, Resource=configmanagements", GroupVersionKind: "configmanagement.gke.io/v1, Kind=ConfigManagement"
Name: "", Namespace: ""
from server for: "config-management-dev.yaml": resource name may not be empty
```

This happens both with the `dev` and `prod` setups. This is the content of one the yaml files:

```
apiVersion: configmanagement.gke.io/v1
kind: ConfigManagement
metadata:
name: config-management
spec:
clusterName: dev
git:
  syncRepo: https://gitlab.com/https://gitlab.com/iht-anthos-test/acm.git
  syncBranch: dev
  secretType: none
policyController:
  enabled: true
```